### DISCUSSION TITLE
Uno Platform JetBrains Rider XAML Hot Reload Timeout issue in Android Physical Device!

### DISCUSSION THREAD

**Discussion by RedEye-Developers** [unoplatform/uno#18763] [Q&A] [answered]
Hi i my Rider All Hot Reload Settings are in Enable include XAML Hot Reload still in Android Physical Device It is Showing Timeout issue I have no idea how to find what is the Problem!

![Screenshot (4)](https://github.com/user-attachments/assets/153ee69f-8df8-4cee-8045-d6fe982a9eab)

**Comment by jeromelaban** [comment]
Thanks for the report. Can you provide a screenshot of the timeout you're mentioning?

**Reply by RedEye-Developers** [reply]
![Screenshot_20241111-184225_FitBites](https://github.com/user-attachments/assets/40191572-ea1d-4bbd-a820-f2893e3e0a2a)

**Reply by jeromelaban** [reply]
Thanks for the report. Are you running on a device or on a simulator? Could you provide logs from the app using:

```cmd
> adb logcat -G 64M
> adb logcat -c
rem Start and crash the app here, wait 2-3 seconds and then:
> adb logcat -d > logcat.txt
```

then post here the `logcat.txt` file?

**Reply by RedEye-Developers** [reply]
i am running in physical Android Device Via USB Cable.

ok i will send a log in few min!

**Reply by RedEye-Developers** [reply]
[logcat.txt](https://github.com/user-attachments/files/17703021/logcat.txt)

**Reply by jeromelaban** [reply]
Thanks. Can you validate again if you temporarily disable the windows firewall on your machine?

**Reply by RedEye-Developers** [reply]
[logcat.txt](https://github.com/user-attachments/files/17703128/logcat.txt)

**Reply by jeromelaban** [reply]
Thanks, you disabled the firewall and it still does not work. Can you set the logging level to Debug in app.xaml.cs? Depending on the template you used it may have different forms.

Can you also tell if it works with an emulator?

**Reply by jeromelaban** [reply]
Can you also show the contents of the Uno Platform tab on the left?

**Reply by RedEye-Developers** [reply]
```cs

        MainWindow = builder.Window;

#if DEBUG
        MainWindow.EnableHotReload();
#endif
        MainWindow.SetWindowIcon();

        Host = await builder.NavigateAsync<Shell>();
    }
```

**Reply by RedEye-Developers** [reply]
![Screenshot (5)](https://github.com/user-attachments/assets/8a867439-7b53-40d9-89cc-633056f38395)

**Reply by jeromelaban** [reply]
Thanks, there's an Uno icon on the left, click on it and show upload the contents of that window with the level set to debug.

**Reply by RedEye-Developers** [reply]
![Screenshot (6)](https://github.com/user-attachments/assets/87d9355e-c704-4810-8576-87975ad64846)

**Reply by RedEye-Developers** [reply]
noting working in that!

**Reply by RedEye-Developers** [reply]
![Screenshot (7)](https://github.com/user-attachments/assets/34d2e761-8ec5-496c-b032-ebfe23c5a2e3)

**Reply by RedEye-Developers** [reply]
![Screenshot (8)](https://github.com/user-attachments/assets/4fe5fc83-a567-46f1-8d77-5017090c4095)

**Reply by jeromelaban** [reply]
Thanks. You need to set the level to Debug first in the Uno Platform log window, then debug your app.

**Reply by RedEye-Developers** [reply]
they have no option for debug in that only have : TRACE, VERBOS, INFO, WARN, ERROR

**Reply by jeromelaban** [reply]
Ah, yes, then set to Trace, the point of the request is to have logs to troubleshoot with :)

**Reply by RedEye-Developers** [reply]
is set that to Trace and not coming any logs when build the application or debugging the application also.

**Reply by RedEye-Developers** [reply]
1>AndroidManifest.xml: Warning XA4218 : Unable to find //manifest/application/uses-library at path: C:\Program Files (x86)\Android\android-sdk\platforms\android-34\optional\androidx.window.extensions.jar
1>AndroidManifest.xml: Warning XA4218 : Unable to find //manifest/application/uses-library at path: C:\Program Files (x86)\Android\android-sdk\platforms\android-34\optional\androidx.window.sidecar.jar

only 2 build error are coming in Build Console and Uno Platform Icon Console is Empty!

**Reply by jeromelaban** [reply]
Thanks. This would explain why hot reload does not work. There are no logs showing at all when setting the trace level? Can you also set the level to trace, then close the solution and reopen it. Are there logs showing then?

**Reply by RedEye-Developers** [reply]
```
Project with Uno.Sdk has been found. Project path C:\Uno Platform Projects\FitBites\FitBites\FitBites.csproj
Generating the csproj.user settings for C:\Uno Platform Projects\FitBites\FitBites\FitBites.csproj.user.
Path to Uno.UI.RemoteControl.Host.dll has been found. Using C:\Users\Administrator\.nuget\packages\uno.winui.devserver\5.5.37\tools\rc\host\net8.0\Uno.UI.RemoteControl.Host.dll
Dev server is started with port 58580.
Creating IDE Channel to Dev Server (751e68d1-e822-4576-8f4f-942c8b3c8952)
Unhandled exception. System.ArgumentException: The provided solution path 'C:\Uno' does not exists
   at Uno.UI.RemoteControl.Host.Program.<>c__DisplayClass0_0.<Main>b__5(String s) in /agent/_work/1/s/src/Uno.UI.RemoteControl.Host/Program.cs:line 49
   at Mono.Options.OptionSet.<>c__DisplayClass22_0.<Add>b__0(OptionValueCollection v)
   at Mono.Options.OptionSet.ActionOption.OnParseComplete(OptionContext c)
   at Mono.Options.Option.Invoke(OptionContext c)
   at Mono.Options.OptionSet.ParseValue(String option, OptionContext c)
   at Mono.Options.OptionSet.Parse(String argument, OptionContext c)
   at Mono.Options.OptionSet.Parse(IEnumerable`1 arguments)
   at Uno.UI.RemoteControl.Host.Program.Main(String[] args) in /agent/_work/1/s/src/Uno.UI.RemoteControl.Host/Program.cs:line 57
   at Uno.UI.RemoteControl.Host.Program.<Main>(String[] args)
```

now it is showing after reopen the solution!

**Reply by jeromelaban** [reply]
Thanks! This is definitely a bug. Could you try moving your project to a folder that has no spaces in it?

**Reply by RedEye-Developers** [reply, marked as answer]
```
Unhandled exception. System.ArgumentException: The provided solution path 'C:\Uno' does not exists
```

Really Really Thanks Bro!

i try 2 full days to fix this issue!

After Moving Entire Uno Project Inside C:\Uno The Hot Reload Timeout issue was Fixed!

**Reply by jeromelaban** [reply]
Thanks for the update! We'll publish a rider update soon that will fix this issue.
